src/User/Application/CommandHandler/RequestPasswordResetCommandHandler.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\User\Application\CommandHandler;
+
+use App\Shared\Domain\Bus\Command\CommandHandlerInterface;
+use App\Shared\Domain\Bus\Event\EventBusInterface;
+use App\User\Application\Command\RequestPasswordResetCommand;
+use App\User\Application\Command\RequestPasswordResetCommandResponse;
+use App\User\Domain\Entity\PasswordResetTokenInterface;
+use App\User\Domain\Entity\UserInterface;
+use App\User\Domain\Event\PasswordResetRequestedEvent;
+use App\User\Domain\Exception\PasswordResetRateLimitExceededException;
+use App\User\Domain\Factory\PasswordResetTokenFactoryInterface;
+use App\User\Domain\Repository\PasswordResetTokenRepositoryInterface;
+use App\User\Domain\Repository\UserRepositoryInterface;
+use Symfony\Component\Uid\Factory\UuidFactory;
+
+final readonly class RequestPasswordResetCommandHandler implements
+    CommandHandlerInterface
+{
+    public function __construct(
+        private UserRepositoryInterface $userRepository,
+        private PasswordResetTokenRepositoryInterface $tokenRepository,
+        private PasswordResetTokenFactoryInterface $tokenFactory,
+        private EventBusInterface $eventBus,
+        private UuidFactory $uuidFactory,
+        private int $rateLimitMaxRequests = 3,
+        private int $rateLimitWindowHours = 1,
+    ) {
+    }
+
+    public function __invoke(RequestPasswordResetCommand $command): void
+    {
+        $successMessage = 'If valid, you will receive a password reset link.';
```

Comment:
please remove the success message, since we don't need to show success message to user for this feature